openedx/features/course_experience/utils.py

```diff
@@ -182,3 +186,16 @@ def get_resume_block(block):
         if resume_block:
             return resume_block
     return block
+
+
+def get_first_purchase_offer_banner_fragment(user, course):
+    if FIRST_PURCHASE_OFFER_BANNER_DISPLAY.is_enabled() and user and course:
+        offer_message = _(u'''{banner_open}15% off your first upgrade.{span_close}
+                               Discount automatically applied.{div_close}''')
+        return Fragment(HTML(offer_message).format(
+            banner_open=HTML(
+                '<div class="first-purchase-offer-banner"><span class="first-purchase-offer-banner-bold">'
```

Comment:
I've noticed we don't seem to set ids on html elements. Seems like adding it would make later optimizely DOM manipulation and writing UI-driven tests easier, but that's just a suggestion.

Reply:
there are front end conventions that recommend only using classes for css as i'm doing here  
and i think usually once an element already has a class there isn't much reason to also add an id